drivers/fpga/fpga_altera.h

```diff
@@ -0,0 +1,207 @@
+#define RECONFIG_DIRECT_COUNT_MASK   0xFF
+
+#define RECONFIG_INDIRECT_ARG_OFFSET 0x08
+#define RECONFIG_INDIRECT_COUNT_MASK 0xFF00
+
+#define RECONFIG_INDIRECT_RESPONSE_OFFSET 0x10
+#define RECONFIG_RESPONSE_COUNT_MASK	  0xFF0000
+
+#define RECONFIG_DATA_MB_CMD_SIZE	   0x10
+#define RECONFIG_DATA_MB_CMD_INDIRECT_MODE 0x01
+
+#define RECONFIG_DATA_MB_CMD_LENGTH 0x03
+
+#define RECONFIG_DATA_MB_CMD_DIRECT_COUNT      0x00
+#define RECONFIG_DATA_MB_CMD_INDIRECT_ARG      0x01
+#define RECONFIG_DATA_MB_CMD_INDIRECT_RESPONSE 0x00
+#define RECONFIG_STATUS_INTERVAL_DELAY_US      1000
+#define RECONFIG_STATUS_RETRY_COUNT            20
+
+#define MBOX_CONFIG_STATUS_STATE_CONFIG	  0x10000000
+#define MBOX_CFGSTAT_VAB_BS_PREAUTH       0x20000000
+
+#define FPGA_NOT_CONFIGURED_ERROR         0x02000004
+
+#define MBOX_CFGSTAT_STATE_ERROR_HARDWARE 0xF0000005
+#define RECONFIG_SOFTFUNC_STATUS_CONF_DONE	  BIT(0)
+#define RECONFIG_SOFTFUNC_STATUS_INIT_DONE	  BIT(1)
+#define RECONFIG_SOFTFUNC_STATUS_SEU_ERROR	  BIT(3)
+#define RECONFIG_PIN_STATUS_NSTATUS		      BIT(31)
+
+#define MBOX_REQUEST_HEADER(cmd_id, cmd_mode, len)                                                 \
+	((cmd_id << MBOX_CMD_CODE_OFFSET) & (MBOX_CMD_ID_MASK)) |                                  \
+		((cmd_mode << MBOX_CMD_MODE_OFFSET) & (MBOX_CMD_MODE_MASK)) |                      \
+		((len << MBOX_DATA_LEN_OFFSET) & (MBOX_DATA_LEN_MASK))
+
+#define MBOX_RECONFIG_REQUEST_DATA_FORMAT(direct_count, indirect_arg_count, response_arg_count)    \
+	((direct_count << RECONFIG_DIRECT_COUNT_OFFSET) & (RECONFIG_DIRECT_COUNT_MASK)) |          \
+		((indirect_arg_count << RECONFIG_INDIRECT_ARG_OFFSET) &                            \
+		 (RECONFIG_INDIRECT_COUNT_MASK)) |                                                 \
+		((response_arg_count << RECONFIG_INDIRECT_RESPONSE_OFFSET) &                       \
+		 (RECONFIG_RESPONSE_COUNT_MASK))
+
+union mailbox_response_header {
+	/* Header of the config status response */
+	uint32_t header;
+
+	struct {
+		/* error_code – Field provides a basic description of whether the command
+		 * succeeded or not. A successful response returns an error code of 0x0,
+		 * non-zero values indicate failure
+		 */
+		uint32_t error_code : 11;
+		/* indirect_bit - Field indicates an indirect command */
+		uint32_t indirect_bit : 1;
+		/* data_length - Field counts the number of word arguments which follow the
+		 * response header word. The meaning of these words depends on the command
+		 * code. Units are words
+		 */
+		uint32_t data_length : 11;
+		/* reserve bit */
+		uint32_t reserved_bit : 1;
+		/* id - Field is returned unchanged from the matching command header and is
+		 * useful for matching responses to commands along with the CLIENT
+		 */
+		uint32_t id : 4;
+		/* client_id - Field is returned unchanged from the matching command header and
+		 * is useful for matching responses to commands along with the ID
+		 */
+		uint32_t client_id : 4;
+	} header_t;
```

Comment:
this is not a typedef so it probably makes sense to remove the `_t`

Reply:
done